pkg/epp/backend/vllm/metrics_test.go

```diff
@@ -17,234 +17,573 @@ limitations under the License.
-							},
-							Gauge: &dto.Gauge{
-								Value: proto.Float64(90),
-							},
-						},
+			},
+			expectedErrCount: 1, // Errors for the two missing metrics
+		},
+		{
+			name: "invalid max lora",
+			metricFamilies: map[string]*dto.MetricFamily{
+				"vllm:lora_requests_info": makeMetricFamily("vllm:lora_requests_info",
+					makeMetric("vllm:lora_requests_info", map[string]string{"running_lora_adapters": "lora1", "max_lora": "invalid"}, 3000.0, 1000),
+				),
+			},
+			mapping: &MetricMapping{
+				LoraRequestInfo: &MetricSpec{MetricName: "vllm:lora_requests_info"},
+			},
+			existingMetrics: &datastore.PodMetrics{
+				Pod: datastore.Pod{
+					Address: "127.0.0.1",
+					NamespacedName: types.NamespacedName{
+						Namespace: "test",
+						Name:      "pod",
 					},
 				},
+				Metrics: datastore.Metrics{},
 			},
-			expectedMetrics: &datastore.Metrics{
-				RunningQueueSize:    15,
-				WaitingQueueSize:    25,
-				KVCacheUsagePercent: 0.9,
-				ActiveModels: map[string]int{
-					"lora3": 0,
-					"lora4": 0,
+			expectedMetrics: &datastore.PodMetrics{
+				Pod: datastore.Pod{
+					Address: "127.0.0.1",
+					NamespacedName: types.NamespacedName{
+						Namespace: "test",
+						Name:      "pod",
+					},
+				},
+				Metrics: datastore.Metrics{
+					ActiveModels:    map[string]int{"lora1": 0},
+					MaxActiveModels: 0, // Should still default to 0.
+
 				},
-				MaxActiveModels: 0,
 			},
-			initialPodMetrics: &datastore.PodMetrics{},
-			expectedErr:       errors.New("strconv.Atoi: parsing '2a': invalid syntax"),
+			expectedErrCount: 1, // Expect *one* error
 		},
 	}
-	for _, tc := range testCases {
+
+	for _, tc := range tests {
 		t.Run(tc.name, func(t *testing.T) {
-			updated, err := promToPodMetrics(logger, tc.metricFamilies, tc.initialPodMetrics)
-			if tc.expectedErr != nil {
-				assert.Error(t, err)
+			p := &PodMetricsClientImpl{MetricMapping: tc.mapping}
+			updated, err := p.promToPodMetrics(logger, tc.metricFamilies, tc.existingMetrics)
+
+			if tc.expectedErrCount == 0 {
+				if err != nil {
+					t.Errorf("promToPodMetrics() unexpected error: %v", err)
+				}
 			} else {
-				assert.NoError(t, err)
-				assert.Equal(t, tc.expectedMetrics, &updated.Metrics)
+				if err == nil {
+					t.Errorf("promToPodMetrics() expected errors, got nil")
+				} else {
+					// Check the *number* of errors.  multierr.Errors() gives us a slice
+					if len(multierr.Errors(err)) != tc.expectedErrCount {
+						t.Errorf("promToPodMetrics() wrong number of errors: got %d, want %d.  Errors: %v", len(multierr.Errors(err)), tc.expectedErrCount, err)
+					}
+
+				}
+			}
+			// Use podMetricsEqual for comparison with tolerance.
+			if !reflect.DeepEqual(updated, tc.expectedMetrics) {
+				t.Errorf("promToPodMetrics() got %+v, want %+v", updated, tc.expectedMetrics)
 			}
 		})
 	}
 }
+
+// TestFetchMetrics is a basic integration test.  A more complete test would mock
+// the HTTP client.
+func TestFetchMetrics(t *testing.T) {
```

Comment:
This test case is justing test an connection error message? I don't we should add this...